shopfloor_single_product_transfer/services/single_product_transfer.py

```diff
@@ -796,6 +796,9 @@ def set_quantity(self, selected_line_id, barcode, quantity, confirmation=False):
         if not move_line.exists():
             # TODO Should probably return to scan_product or scan_location?
             return self._response_for_set_quantity(move_line)
+        if move_line.state == "done":
```

Comment:
We need to be sure the line is not changed to done during the whole transaction
```suggestion
        self._lock_lines(move_line)
        if move_line.state == "done":
```